Helper/Cart.php

```diff
@@ -1304,7 +1304,7 @@ protected function getCalculationAddress($quote)
      * @return array
      * @throws \Exception
      */
-    public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote = null)
+    public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote = null, $requireBillingAddress = true)
```

Comment:
@dumega Taking into account the complexity of the problem and it's root case on Bolt end - not calling shipping and tax for virtual carts and inconsistency in processing guest vs. logged in user - discussed in our private slack channel, and having in mind that this PR only solves an edge case, leaving the main problem open, I'd like to suggest one simplification here that you may or may not consider.

From Bolt perspective there's only one address, `shipping == billing`. Although billing address is not present in every request type data, it is the same. So, I'll remove logic around `$requireBillingAddress` (everywhere) and use shipping address as billing one everywhere billing is missing. I've checked how `$requireBillingAddress` is used in the code and it does not make much sense.